auraescript/src/lib.rs

```diff
@@ -164,12 +166,15 @@ impl ModuleLoader for TypescriptModuleLoader {
         module_specifier: &ModuleSpecifier,
         _maybe_referrer: Option<&ModuleSpecifier>,
         _is_dyn_import: bool,
-    ) -> Pin<Box<ModuleSourceFuture>> {
-        let module_specifier = module_specifier.clone();
-        async move {
+        _requested_module_type: RequestedModuleType,
+    ) -> ModuleLoadResponse {
+        fn load(
+            source_maps: SourceMapStore,
+            module_specifier: &ModuleSpecifier,
+        ) -> Result<ModuleSource, Error> {
             let path = module_specifier
                 .to_file_path()
-                .map_err(|_| anyhow!("Only file: URLs are supported."))?;
+                .map_err(|_| anyhow!("Only file:// URLs are supported."))?;
```

Comment:
should we use the requested module type instead of extracting it from the path?  or maybe use the requested module path and the path-derived version of None is passed in?